pkg/metrics/metrics.go

```diff
@@ -173,13 +193,77 @@ func (m *Metrics) UpdateCertificateExpiry(crt *v1alpha1.Certificate, secretListe
 		return
 	}
 
-	updateX509Expiry(crt.Name, crt.Namespace, cert)
+	updateX509Expiry(crt, cert)
 }
 
-func updateX509Expiry(name, namespace string, cert *x509.Certificate) {
-	// set certificate expiry time
+func updateX509Expiry(crt *v1alpha1.Certificate, cert *x509.Certificate) {
 	expiryTime := cert.NotAfter
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(crt)
+	if err != nil {
+		return
+	}
+
+	registeredCertificates.mtx.Lock()
+	defer registeredCertificates.mtx.Unlock()
+	// set certificate expiry time
 	CertificateExpiryTimeSeconds.With(prometheus.Labels{
-		"name":      name,
-		"namespace": namespace}).Set(float64(expiryTime.Unix()))
+		"name":      crt.Name,
+		"namespace": crt.Namespace}).Set(float64(expiryTime.Unix()))
+	registeredCertificates.certificates[key] = struct{}{}
```

Comment:
We should move the logic to set this var into its own function in future, as we'll likely add more Certificate-level metrics (e.g. #1679) and we shouldn't be duplicating this logic. Can be done with #1679 😄